app/league/[leagueId]/entry/[entryId]/week/Week.tsx

```diff
@@ -110,44 +125,59 @@ const Week = ({ entry, league, NFLTeams, week }: IWeekProps): JSX.Element => {
   };
 
   useEffect(() => {
+    if (!selectedLeague) {
+      getSelectedLeague();
+      return;
+    }
     getSchedule(week);
-  }, [week]);
+    setIsLoading(false);
+  }, [week, selectedLeague]);
```

Comment:
By adding `selectedLeague`, it's going to cause side effects. Every time selectedLeague() is called, it will run this useEffect.